test/extended/prometheus/prometheus.go

```diff
@@ -42,6 +44,225 @@ import (
 	helper "github.com/openshift/origin/test/extended/util/prometheus"
 )
 
+var _ = g.Describe("[sig-instrumentation][Late] OpenShift alerting rules", func() {
+	defer g.GinkgoRecover()
+
+	// These alerts are known to be missing the summary and/or description
+	// labels.  Bugzillas are being filed, and will be linked here.  These
+	// should be fixed one-by-one and removed from this list.
+	descriptionExceptions := sets.NewString(
+		"APIRemovedInNextEUSReleaseInUse",
+		"APIRemovedInNextReleaseInUse",
+		"CertifiedOperatorsCatalogError",
+		"CloudCredentialOperatorDeprovisioningFailed",
+		"CloudCredentialOperatorInsufficientCloudCreds",
+		"CloudCredentialOperatorProvisioningFailed",
+		"CloudCredentialOperatorTargetNamespaceMissing",
+		"ClusterProxyApplySlow",
+		"CommunityOperatorsCatalogError",
+		"CsvAbnormalFailedOver2Min",
+		"CsvAbnormalOver30Min",
+		"ExtremelyHighIndividualControlPlaneCPU",
+		"HAProxyDown",
+		"HAProxyReloadFail",
+		"HighOverallControlPlaneCPU",
+		"ImageRegistryStorageReconfigured",
+		"IngressControllerDegraded",
+		"IngressControllerUnavailable",
+		"InstallPlanStepAppliedWithWarnings",
+		"KubeControllerManagerDown",
+		"KubeSchedulerDown",
+		"KubeletHealthState",
+		"MCDDrainError",
+		"MCDPivotError",
+		"MCDRebootError",
+		"MachineAPIOperatorMetricsCollectionFailing",
+		"MachineApproverMaxPendingCSRsReached",
+		"MachineHealthCheckUnterminatedShortCircuit",
+		"MachineNotYetDeleted",
+		"MachineWithNoRunningPhase",
+		"MachineWithoutValidNode",
+		"MasterNodesHighMemoryUsage",
+		"NodeProxyApplySlow",
+		"NodeProxyApplyStale",
+		"NodeWithoutSDNPod",
+		"PodDisruptionBudgetAtLimit",
+		"PodDisruptionBudgetLimit",
+		"RedhatMarketplaceCatalogError",
+		"RedhatOperatorsCatalogError",
+		"SDNPodNotReady",
+		"SamplesDegraded",
+		"SamplesImagestreamImportFailing",
+		"SamplesInvalidConfig",
+		"SamplesMissingSecret",
+		"SamplesMissingTBRCredential",
+		"SamplesRetriesMissingOnImagestreamImportFailing",
+		"SamplesTBRInaccessibleOnBoot",
+		"SchedulerLegacyPolicySet",
+		"SystemMemoryExceedsReservation",
+		"TechPreviewNoUpgrade",
+		"etcdBackendQuotaLowSpace",
+		"etcdExcessiveDatabaseGrowth",
+		"etcdHighFsyncDurations",
+	)
+
+	var alertingRules map[string][]alertingRule
+	oc := exutil.NewCLIWithoutNamespace("prometheus")
+
+	g.BeforeEach(func() {
+		url, _, bearerToken, ok := helper.LocatePrometheus(oc)
+		if !ok {
+			e2e.Failf("Prometheus could not be located on this cluster, failing prometheus test")
+		}
+
+		if alertingRules == nil {
+			var err error
```

Comment:
I was trying to avoid fetching these each time, while still being able to separate each check into its own `g.It("should..")` block. So `alertingRules` is actually defined in the outer scope, and it only does the fetch if it hasn't been initialized. Having the call to `fetchAlertingRules()` initialize `err` would shadow `alertingRules`. Let me know if I'm misunderstanding though.

Reply:
Ah indeed, I missed that alertingRules was defined in the outer scope, makes sense now :+1: